Source/FakeItEasy/Core/DefaultFakeManagerAccessor.cs

```diff
@@ -23,11 +22,20 @@ public FakeManager GetFakeManager(object proxy)
 
             var taggable = AsTaggable(proxy);
 
+            if (taggable.Tag == null)
+            {
+                throw new ArgumentException("The specified object does not have a FakeManager attached.");

```

Comment:
Agreed, @adamralph. Check out my [screed in the conversation tab](https://github.com/FakeItEasy/FakeItEasy/pull/538#issuecomment-141657679) for this and other concerns!